pkg/migration/migrator.go

```diff
@@ -67,6 +68,51 @@ type VersionComparator func(string, string) bool
 // Known errors which we can ignore
 var ErrAlreadyUpToDate = errors.New("already up to date")
 
+// checkVersionCompatibility validates that the context version is supported by the current CLI
+func checkVersionCompatibility(contextVersion, latestSupported, contextFile string) error {
+	if contextVersion == "" {
+		// Missing version - could be very old context, warn but allow
+		return fmt.Errorf("context file %s is missing version field - this may be an old context that needs migration", contextFile)
+	}
+
+	// For CLI version v0.0.8, we can support up to context version 0.0.5
+	// Override the latestSupported if this CLI version supports higher
+	cliVersion := version.GetVersion()
+	if cliVersion == "v0.0.8" || cliVersion == "0.0.8" {
+		latestSupported = "0.0.5"
+	}
+	if cliVersion == "v0.0.7" || cliVersion == "0.0.7" {
+		latestSupported = "0.0.3"
+	}
+	if cliVersion == "v0.0.6" || cliVersion == "0.0.6" {
+		latestSupported = "0.0.3"
+	}
+	if cliVersion == "v0.0.4" || cliVersion == "0.0.4" {
+		latestSupported = "0.0.3"
+	}
+	if cliVersion == "v0.0.3" || cliVersion == "0.0.3" {
+		latestSupported = "0.0.3"
+	}
+	if cliVersion == "v0.0.2" || cliVersion == "0.0.2" {
+		latestSupported = "0.0.3"
+	}
+	if cliVersion == "v0.0.1" || cliVersion == "0.0.1" {
+		latestSupported = "0.0.1"
+	}
+
+	// If context version is newer than what we support, return compatibility error
+	if versionGreaterThan(contextVersion, latestSupported) {
+		return &common.VersionCompatibilityError{
+			ContextVersion:  contextVersion,
+			CLIVersion:      version.GetVersion(),
```

Comment:
We already have cliVersion in context:

```suggestion
			CLIVersion:      cliVersion,
```